### PR TITLE
Fix: Added support for show_api in mount_gradio_app

### PR DESCRIPTION
**Pull request by HongweiRuan**
Fix: Added support for show_api in mount_gradio_app

## Description
The `mount_gradio_app` function did not previously support the `show_api` parameter, which caused the "Use via API" button to always appear, regardless of the user's intent.

### Changes
- Added logic to handle `show_api` in `mount_gradio_app`.
- Ensured `show_api` is correctly passed to the `Blocks` instance.

Closes: #9980

### CODE REVIEW

**Comment by HongweiRuan** [comment]
Hi @freddyaboulton , could you please approve the workflows for this PR? Thanks!

**Comment by abidlabs** [comment]
The `app_kwargs` parameter is meant for "additional keyword arguments to pass to the underlying FastAPI app" -- since `show_api` is actually an attribute of Blocks, not FastAPI, I would actually add it as a separate argument in `mount_gradio_app`

**Comment by HongweiRuan** [comment]
Hi @abidlabs,

Thank you for the clarification! This is my very first time trying to make contributions. The original issue described the problem as follows:

"I create a Gradio app using mount_gradio_app so I don't need to use the launch function call. I want to hide the "Use via API" on the footer. **I try to add app_kwargs={"show_api": False} when calling the mount_gradio_app()** and the "Use via API" message is still shown on the footer."

Based on this description, I initially thought the user wanted to achieve this functionality via app_kwargs={"show_api": False}, which led to my initial implementation.

I now understand from your explanation that show_api is an attribute of Blocks, not FastAPI. I’m happy to continue fixing this issue following your suggestion to add show_api as a separate argument in mount_gradio_app.

**Comment by abidlabs** [comment]
Yes thanks for the explanation @HongweiRuan please feel free to make any modifications to this PR and we can review

**Comment by abidlabs** [comment]
@HongweiRuan are you able to tweak your PR based on my suggestion above?

**Comment by HongweiRuan** [comment]
> @HongweiRuan are you able to tweak your PR based on my suggestion above?

Hi @abidlabs, sorry I was preparing for my final exams these days and haven't got a time to fix it, can I fix it in several days?

**Comment by HongweiRuan** [comment]
> Yea should be fine @HongweiRuan . We just don't want to leave forgotten PRs open or stale!

Yes, I will definitely look into it when I got some time.

**Comment by HongweiRuan** [comment]
Hi @abidlabs @freddyaboulton, could you check my newest commit please? I think that fixes the problem, thank you!

**Comment by HongweiRuan** [comment]
Hi @abidlabs @freddyaboulton ,sorry there was one blank line contained a whitespace in my code causes one test to fail, I have fixed that and should be good to test again!

File: gradio/routes.py [hunk 1]
```diff
@@ -1524,6 +1524,7 @@ def mount_gradio_app(
     path: str,
     server_name: str = "0.0.0.0",
     server_port: int = 7860,
+    show_api: bool = True,
```
Comment (abidlabs) [inline gradio/routes.py:1527]:
```suggestion
    show_api: bool | None = None,
```

File: gradio/routes.py [hunk 2]
```diff
@@ -1578,6 +1580,7 @@ def read_main():
         )
 
     blocks.dev_mode = False
+    blocks.show_api = show_api
```
Comment (abidlabs) [inline gradio/routes.py:1583]:
```suggestion
    if show_api is not None:
        blocks.show_api = show_api
```

**Review by abidlabs — APPROVED** [review]
lgtm thanks @HongweiRuan!

**Comment by HongweiRuan** [comment]
@abidlabs Thank you very much for your support and patience on my first contribution!

**Pull request merged**